client/components/tinymce/index.jsx

```diff
@@ -416,9 +416,11 @@ module.exports = React.createClass( {
 		if ( this._editor ) {
 			const { mode } = this.props;
 			this._editor.setContent( wpautop( content ), { ...args, mode } );
-
-			// clear the undo stack to ensure that we don't have any leftovers
-			this._editor.undoManager.clear();
+			const resetHistory = args && args.initial;
```

Comment:
Thinking we could avoid the truthy check by either providing a default value for the argument:

```js
setEditorContent( content, args = {} )
```

Or by assigning the merged args above into a variable:

```js
args = { ...args, mode };
this._editor.setContent( wpautop( content ), args );
```

At which point we could update this to call on `initial` directly:

```js
if ( args.initial ) {
	// Clear the undo stack when initially setting content
	this._editor.undoManager.clear();
}
```

What do you think?